### DISCUSSION TITLE
Call for Wi-Fi CSI Application Requirements

### DISCUSSION THREAD

**Discussion by MacChu0315-Espressif** [espressif/esp-csi#120] [General]
# Call for Wi-Fi CSI Application Requirements

Hello everyone, thank you for your attention and contribution to this project! In order to better promote the development of this project, we have opened the Discussion Board and are now soliciting opinions on the follow-up development of this project.

## Background

Channel state information (CSI) includes specific indicators such as carrier signal strength, amplitude, phase, and signal delay. These indicators reveal the signal scattering, reflection, and power attenuation phenomenon that occur with the carrier as the transmission distance changes. It can be used to measure the channel status of the wireless network in Wi-Fi communication. By analyzing and studying the changes in CSI, we can conversely speculate on the changes in the physical environment that cause the changes in the channel state, that is, to achieve non-contact intelligent sensing.

## Possible Application Requirements

Based on the progress achieved, we consider some possible application scenarios and analyze the different requirements for CSI technology in different application scenarios:

1. Human fall detection:
   - Fall detection technology plays a vital role in the elderly and people with reduced mobility, enabling timely detection and response to fall events to reduce the potential risk of injury.
   - In this application scenario, the CSI device is more like the role of the traditional medium millimeter wave radar. In order to detect actions that do not last long, CSI devices are required to work at a high and stable sampling frequency. This means that it will be less easy for a CSI device to turn on the wireless function at the same time, and it also means that it will be almost impossible for it to act as a normal wireless node at the same time.

2. Smart home:
   - Using wireless sensing technology, smart homes can realize intelligent energy management, such as automatically sensing the presence or absence of people indoors and their activity status (such as sleep status), adjusting the use of lighting and air conditioning to achieve energy saving and environmental protection.
   - In this application scenario, the CSI device will share RF module for CSI sensing and other wireless connections such as BLE and MQTT. Therefore, in this state, the CSI device will only work at a lower sampling rate. A challenge in this case is that the complex indoor environment may negatively affect the CSI detection accuracy.

3. Unmanned factory:
   - Since the unmanned factory is a scene where there is no movement or only regular movement for a long time, we can accurately detect the occurrence of unexpected behavior by filtering out the long-term regularity signal in the CSI information. In this case, in addition to the requirements for the algorithm, there are also higher requirements for the stability of the chip.

## Call for Ideas

- We believe our consideration is still one-sided and insufficient, so we hope to collect your ideas and brainstorm: For example, are there other possible application scenarios that we have overlooked? How do you consider the priorities of different application scenarios? Are there any specific requirements or suggestions for the realization of these application scenarios?
- We welcome developers to experience our repository and share feedback on requirements and technology. If you are a relevant practitioner or researcher, we sincerely hope that our technology can better serve your application or research. Therefore, we look forward to your more specific application requirements, and hope that we can work together to advance your work.
- We hope everyone can actively participate and tell us your thoughts. You are welcome to leave a message in the comment area below and please feel free to [contact us](https://www.espressif.com/zh-hans/contact-us/sales-questions) if you have any business questions.

**Comment by HavingaThijs** [comment]
CSI extraction is indeed very valuable. One other application would be interference detection, especially narrowband interference like Bluetooth or IEEE 802.15.4 (ZigBee, Thread). However, this requires being able to extract the CSI even when the data or even (part of the) preamble decoding fails. I'm wondering if this is possible with the current ESP chips? It looks like the `rx_state` is always 0 (meaning success) when getting the CSI via the Rx callback.

**Comment by MacChu0315-Espressif** [comment]
Hi @HavingaThijs, thank you for your insightful comment :)
While CSI extraction is indeed a valuable tool, it might not always be the most straightforward approach for network environment analysis.  In many cases, simple tools like ping can provide the necessary insights more efficiently.
BTW, you can look forward to the capabilities of the upcoming ESP32-C5.  This will be Espressif's first SoC to support dual-band Wi-Fi 6 in both 2.4 GHz and 5 GHz frequencies, along with connectivity features such as Bluetooth 5 (LE) and IEEE 802.15.4 (Zigbee 3.0, Thread 1.3).
I hope this helps.  If you have any further questions or needs, please feel free to contact us again.
